packages/sdk/src/controllers/DataConnectorController.ts

```diff
@@ -0,0 +1,112 @@
+     * @param config page configuration
+     * @param context context that will be available in the connector script.
+     * @returns a DataPage with an array of data objects
+     */
+    getPage = async (connectorId: string, config: PageConfig, context: MetaData = {}) => {
+        const res = await this.#editorAPI;
+        return res
+            .dataConnectorGetPage(connectorId, JSON.stringify(config), JSON.stringify(context))
+            .then((result) => getEditorResponseData<DataPage>(result))
+            .then((resp) => {
+                const update: EditorResponse<DataPage> = { ...resp, parsedData: null };
+                if (resp.parsedData) {
+                    update.parsedData = {
+                        data: resp.parsedData.data.map((e: DataItem) => this.parseDateProperties(e)),
+                        continuationToken: resp.parsedData.continuationToken,
+                    };
+                }
+                return update;
+            });
+    };
+
+    /**
+     * Query a specific DataConnector for a data model using the dynamic context as parameter.
+     * @param connectorId unique id of the media connector
+     * @param context context that will be available in the connector script.
+     * @returns DataModel
+     */
+    getModel = async (connectorId: string, context: MetaData = {}) => {
+        const res = await this.#editorAPI;
+        return res
+            .dataConnectorGetModel(connectorId, JSON.stringify(context))
+            .then((result) => getEditorResponseData<ConnectorConfigOptions>(result));
+    };
+
+    /**
+     * All connectors have a certain set of mappings they allow to be passed into the connector methods their context. This
+     * method allows you to discover which mappings are available for a given connector. If you want to use any of these
+     * mappings, they will be available in the `context` parameter of any connector method.
+     * @param connectorId unique id of the media connector
+     * @returns connector mappings
+     */
+    getConfigurationOptions = async (connectorId: string) => {
+        const res = await this.#editorAPI;
+        return res
+            .dataConnectorGetConfigurationOptions(connectorId)
+            .then((result) => getEditorResponseData<ConnectorConfigOptions>(result));
+    };
+
+    /**
+     * This method returns what capabilities the selected connector has. It gives an indication what methods can
+     * be used successfully for a certain connector.
+     * @param connectorId unique id of the Data connector
+     * @returns DataConnectorCapabilities
+     */
+    getCapabilities = async (connectorId: string) => {
+        const res = await this.#editorAPI;
+        return res
+            .dataConnectorGetCapabilities(connectorId)
+            .then((result) => getEditorResponseData<DataConnectorCapabilities>(result));
+    };
+
+    private parseDateProperties(dataItem: DataItem): DataItem {
```

Comment:
Won't there be an issue here when the connector package is trying to run the connector in standalone and what is returned in the page object won't be parsed as a date since it is done only via the SDK!?

Reply:
I think there should be a discussion regarding that matter, if we need to include this parsing method to the `DataConnectorContext`, am I right @pietervp ?

Reply:
Yeah I'll schedule a quick sync about this